docs/tutorials/contributing/strategy/writing_test_for_the_new_strategy.rst

```diff
@@ -65,66 +92,60 @@ All three of these functions can take an optional keyword argument
 
    This is equivalent to::
 
-    def test_effect_of_strategy(self):
-        P1 = axelrod.TitForTat() # Or whatever player is in your test class
-        P2 = axelrod.Player()
-        P2.history = []
-        P2.history = []
-        self.assertEqual(P1.strategy(P2), 'C')
+    P1 = axelrod.TitForTat() # Or whatever player is in your test class
+    P2 = axelrod.Player()
+    self.assertEqual(P1.strategy(P2), 'C')
 
-2. The member function :code:`markov_test` takes a list of four plays, each
+2. The member function :code:`second_play_test` takes a list of four plays, each
    following one round of CC, CD, DC, and DD respectively::
 
-    def test_effect_of_strategy(self):
-        self.markov_test(['C', 'D', 'D', 'C'])
+    self.second_play_test('C', 'D', 'D', 'C')
 
-   This is equivalent to::
-
-    def test_effect_of_strategy(self):
-        P1 = axelrod.TitForTat() # Or whatever player is in your test class
-        P2 = axelrod.Player()
-        P2.history = ['C']
-        P2.history = ['C']
-        self.assertEqual(P1.strategy(P2), 'C')
-        P2.history = ['C']
-        P2.history = ['D']
-        self.assertEqual(P1.strategy(P2), 'D')
-        P2.history = ['D']
-        P2.history = ['C']
-        self.assertEqual(P1.strategy(P2), 'D')
-        P2.history = ['D']
-        P2.history = ['D']
-        self.assertEqual(P1.strategy(P2), 'C')
+   This is equivalent to choosing an opponent will play C or D as needed and
+   checking the next move. This function can also take an optional random seed
+   argument `seed`.
 
 3. The member function :code:`responses_test` takes arbitrary histories for each
    player and tests a list of expected next responses::
 
     def test_effect_of_strategy(self):
-        self.responses_test([C], [C], [D, C, C, C], random_seed=15)
+        self.responses_test([D, C, C, C], [C], [C], random_seed=15)
+
+   In this case each player has their history simulated to  be :code:`[C]` and
+   the expected responses are D, C, C, C. Note that the histories will elongate
+   as the responses accumulated, with the opponent accruing cooperations.
 
-   In this case each player has their history set to :code:`[C]` and the
-   expected responses are D, C, C, C. Note that the histories will elongate as
-   the responses accumulated.
+   If the given histories are not possible for the strategy then the test will
+   not be meaningful. For example, setting the history of Defector to have
+   cooperations is not a possible history of play since Defector always defects,
+   and so will not actually test the strategy correctly. The test suite will
+   warn you if it detects a mismatch in simulated history and actual history.
+
+   Note also that in general it is not a good idea to manually set the history
```

Comment:
Yes I agree with what you say in the sentence (because I know why which the reader might not) but with this framework there's not much need for the sentence (as people are unlikely to do it right?) or at least you could add the explanation?